lib/chef/property.rb

```diff
@@ -748,5 +749,22 @@ def coerce_and_validate(resource, value)
 
       result
     end
+
+    # recursively dup the value
+    def deep_dup(value)
```

Comment:
Do we really not have a deep_dup somewhere? We do this in the Node constnatly, right?

Reply:
That is all overly complicated stuff hanging off of the various attribute classes, which would need to be injected into Hash and Array to function similarly.